types/lodash/ts3.1/common/array.d.ts

```diff
@@ -488,6 +488,7 @@ declare module "../index" {
          */
         flatten(): T extends Many<infer U> ? CollectionChain<U> : CollectionChain<T>;
     }
+    type Flat<T> = (T extends List<infer U> ? never : T);
```

Comment:
Why is this needed for chains but not other versions?

Also don't `infer U` if you never use it. Instead just use `any`.

Reply:
I've switched `infer U` to `any`.  

I didn't realize that the static version wasn't working until I noticed that the tests were all asserting the final type via a generic.  I've added this workaround to the `LodashStatic` variant of `_.flattenDeep()` and updated the tests to add cases where the explicit generics are not provided.